tests/lwe_to_polynomial/sk_encrypt_op.mlir

```diff
@@ -0,0 +1,32 @@
+// RUN: heir-opt %s --lwe-to-polynomial | FileCheck %s
+
+#encoding = #lwe.polynomial_coefficient_encoding<cleartext_start=15, cleartext_bitwidth=15>
+#my_poly = #polynomial.int_polynomial<1 + x**1024>
+#ring = #polynomial.ring<coefficientType = i32, coefficientModulus = 7917 : i32, polynomialModulus=#my_poly>
+#rlwe_params = #lwe.rlwe_params<dimension=2, ring=#ring>
+!plaintext_rlwe = !lwe.rlwe_plaintext<encoding = #encoding, ring = #ring, underlying_type=i3>
+!ciphertext_rlwe = !lwe.rlwe_ciphertext<encoding = #encoding, rlwe_params = #rlwe_params, underlying_type=i3>
+!rlwe_privkey = !lwe.rlwe_secret_key<rlwe_params=#rlwe_params>
+
+func.func @test_rlwe_encrypt(%arg0: !plaintext_rlwe, %arg1: !rlwe_privkey) -> !ciphertext_rlwe {
+  // CHECK-NOT: lwe.rlwe_encrypt
+
+  // CHECK-DAG: %[[ZERO:.*]] = arith.constant 0 : index
+
+  // CHECK-DAG: %[[PRNG:.*]] = random.init_prng %[[SEED:.*]]
+  // CHECK-DAG: %[[UNIFORM:.*]] = random.discrete_uniform_distribution %[[PRNG]]
+  // CHECK-DAG: %[[GAUSSIAN:.*]] = random.discrete_gaussian_distribution %[[PRNG]]
+
+  // CHECK-DAG: %[[SK:.*]] = tensor.extract %arg1[%[[ZERO]]]
+  // CHECK-DAG: %[[SAMPLE_U:.*]] = random.sample %[[UNIFORM]]
+  // CHECK-DAG: %[[U:.*]] = polynomial.from_tensor %[[SAMPLE_U]]
+  // CHECK: %[[SAMPLE_E:.*]] = random.sample %[[GAUSSIAN]]
+  // CHECK: %[[E:.*]] = polynomial.from_tensor %[[SAMPLE_E]]
+  // CHECK-DAG: %[[SK_TIMES_U:.*]] = polynomial.mul %[[SK]], %[[U]]
```

Comment:
When I run the test, it doesn't pass for me. Here are some changes I made for it to pass on my machine:
```
  // CHECK-DAG: %[[SAMPLE_E:.*]] = random.sample %[[GAUSSIAN]]
  // CHECK-DAG: %[[E:.*]] = polynomial.from_tensor %[[SAMPLE_E]]
  // CHECK-DAG: %[[SK_TIMES_U:.*]] = polynomial.mul %[[U]], %[[SK]]
```
Unfortunately, SAMPLE_E and E come in a different order than what the test currently requires. Hence, I modified them to be CHECK-DAG's instead of CHECK. 

Also, I noticed that the polynomial multiplication multiplies U by SK instead of SK by U, so I've reordered them. Now it passes for me.

Reply:
Should probably rename `SK_TIMES_U` to `U_TIMES_SK` for naming consistency

Reply:
Yeah the tests were still failing at the point of the last commit - I got them fixed up tho and committed them, but unfortunately I messed up a bit on the rebasing and included some extraneous commits, which I will now remove

Reply:
Ok ended up fixing the commit history ... separated the tests into 2 files to keep the FileCheck Naming a bit more legible. lmk if you want me to change anything with the sk_encrypt_op.mlir file or the pk_encrypt_op.mlir file.